update Jenkinsfile and fix Android/iOS usage

AlwinEsch opened the pull request.

AlwinEsch committed:
update Jenkinsfile

Rechi commented:
Tested on osx and android, on android it doesn't work.

AlwinEsch commented:
Was a ERROR on Log?

Rechi commented:
#### Error
AddOnLog: CpBlobs: failed to create SOIL texture '/storage/emulated/0/Android/data/org.xbmc.kodi/files/.kodi/addons/screensaver.cpblobs/resources/cube.dds', SOIL error 'No cube map capability present'


#### Device info
GL_VENDOR = Qualcomm
GL_RENDERER = Adreno (TM) 540
GL_VERSION = OpenGL ES 3.2 V@331.0 (GIT@b1733ef, I8d805acd35) (Date:10/30/18)
GL_SHADING_LANGUAGE_VERSION = OpenGL ES GLSL ES 3.20
GL_EXTENSIONS = GL_OES_EGL_image GL_OES_EGL_image_external GL_OES_EGL_sync GL_OES_vertex_half_float GL_OES_framebuffer_object GL_OES_rgb8_rgba8 GL_OES_compressed_ETC1_RGB8_texture GL_AMD_compressed_ATC_texture GL_KHR_texture_compression_astc_ldr GL_KHR_texture_compression_astc_hdr GL_OES_texture_compression_astc GL_OES_texture_npot GL_EXT_texture_filter_anisotropic GL_EXT_texture_format_BGRA8888 GL_OES_texture_3D GL_EXT_color_buffer_float GL_EXT_color_buffer_half_float GL_QCOM_alpha_test GL_OES_depth24 GL_OES_packed_depth_stencil GL_OES_depth_texture GL_OES_depth_texture_cube_map GL_EXT_sRGB GL_OES_texture_float GL_OES_texture_float_linear GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_EXT_texture_type_2_10_10_10_REV GL_EXT_texture_sRGB_decode GL_EXT_texture_format_sRGB_override GL_OES_element_index_uint GL_EXT_copy_image GL_EXT_geometry_shader GL_EXT_tessellation_shader GL_OES_texture_stencil8 GL_EXT_shader_io_blocks GL_OES_shader_image_atomic GL_OES_sample_variables GL_EXT_texture_border_clamp GL_EXT_EGL_image_external_wrap_modes GL_EXT_multisampled_render_to_texture GL_EXT_multisampled_render_to_texture2 GL_OES_shader_multisample_interpolation GL_EXT_texture_cube_map_array GL_EXT_draw_buffers_indexed GL_EXT_gpu_shader5 GL_EXT_robustness GL_EXT_texture_buffer GL_EXT_shader_framebuffer_fetch GL_ARM_shader_framebuffer_fetch_depth_stencil GL_OES_texture_storage_multisample_2d_array GL_OES_sample_shading GL_OES_get_program_binary GL_EXT_debug_label GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent GL_QCOM_tiled_rendering GL_ANDROID_extension_pack_es31a GL_EXT_primitive_bounding_box GL_OES_standard_derivatives GL_OES_vertex_array_object GL_EXT_disjoint_timer_query GL_KHR_debug GL_EXT_YUV_target GL_EXT_sRGB_write_control GL_EXT_texture_norm16 GL_EXT_discard_framebuffer GL_OES_surfaceless_context GL_OVR_multiview GL_OVR_multiview2 GL_EXT_texture_sRGB_R8 GL_KHR_no_error GL_EXT_debug_marker GL_OES_EGL_image_external_essl3 GL_OVR_multiview_multisampled_render_to_texture GL_EXT_buffer_storage GL_EXT_external_buffer GL_EXT_blit_framebuffer_params GL_EXT_clip_cull_distance GL_EXT_protected_textures GL_EXT_shader_non_constant_global_initializers GL_QCOM_texture_foveated GL_QCOM_shader_framebuffer_fetch_noncoherent GL_EXT_memory_object GL_EXT_memory_object_fd GL_EXT_EGL_image_array GL_NV_shader_noperspective_interpolation GL_KHR_robust_buffer_access_behavior GL_EXT_EGL_image_storage GL_EXT_blend_func_extended GL_EXT_clip_control

AlwinEsch commented:
With the second commit it seems to be fixed:
![IMG_0003](https://user-images.githubusercontent.com/6879739/54708146-02993d80-4b43-11e9-84d4-260d744c7a69.jpg)

Rechi commented:
works now on Android
tested it on an iPhone 4S and there it doesn't work (same error)

AlwinEsch commented:
How is there the Device Info?

AlwinEsch commented:
Have also temporary changed the `query_cubemap_capability()` to give always present, to know that maybe one have a extension can bring it.

AlwinEsch commented on CMakeLists.txt:
```diff
@@ -35,4 +35,5 @@ list(APPEND DEPLIBS soil2)
 
 build_addon(screensaver.cpblobs CPBLOBS DEPLIBS)
 
+#message(FATAL_ERROR "-11--- ${CMAKE_CXX_STANDARD}   -------- ${CXX11_SWITCH}")
```
Forget this :-D (was only from a test of a request from Kodi)

Rechi commented:
It works with the always return present variant.

#### Device info iPhone 4S
GL_VENDOR = Imagination Technologies
GL_RENDERER = PowerVR SGX 543
GL_VERSION = OpenGL ES 2.0 IMGSGX543-124.1
GL_SHADING_LANGUAGE_VERSION = OpenGL ES GLSL ES 1.00
GL_EXTENSIONS = GL_OES_depth_texture GL_OES_depth24 GL_OES_element_index_uint GL_OES_fbo_render_mipmap GL_OES_mapbuffer GL_OES_packed_depth_stencil GL_OES_rgb8_rgba8 GL_OES_standard_derivatives GL_OES_texture_float GL_OES_texture_half_float GL_OES_texture_half_float_linear GL_OES_vertex_array_object GL_EXT_blend_minmax GL_EXT_color_buffer_half_float GL_EXT_debug_label GL_EXT_debug_marker GL_EXT_discard_framebuffer GL_EXT_draw_instanced GL_EXT_instanced_arrays GL_EXT_map_buffer_range GL_EXT_occlusion_query_boolean GL_EXT_pvrtc_sRGB GL_EXT_read_format_bgra GL_EXT_separate_shader_objects GL_EXT_shader_framebuffer_fetch GL_EXT_shader_texture_lod GL_EXT_shadow_samplers GL_EXT_sRGB GL_EXT_texture_filter_anisotropic GL_EXT_texture_rg GL_EXT_texture_storage GL_APPLE_clip_distance GL_APPLE_color_buffer_packed_float GL_APPLE_copy_texture_levels GL_APPLE_framebuffer_multisample GL_APPLE_rgb_422 GL_APPLE_sync GL_APPLE_texture_format_BGRA8888 GL_APPLE_texture_max_level GL_APPLE_texture_packed_float GL_IMG_read_format GL_IMG_texture_compression_pvrtc

AlwinEsch committed:
fix `query_cubemap_capability()` on Android where returns *_NONE

It report not supported "GL_EXT_texture_cube_map" but supported
"GL_EXT_texture_cube_map_array". By test with them the addon worked
normal.

AlwinEsch committed:
fix `query_cubemap_capability()` on iOS where returns *_NONE

It report not supported "GL_EXT_texture_cube_map" and not supported
"GL_EXT_texture_cube_map_array". But by test the addon worked
normal with this check prevented.

AlwinEsch commented:
Have made the Temp to a right commit, there on iOS extension is nothing visible which can be interpreted as cubemap support. Is a bit hacky then, but if it works, why not ;-).

@Rechi for you OK?

Rechi commented:
Still works on the old iPhone 4S.

AlwinEsch commented:
Thanks a lot for testing!
